### ISSUE TITLE
Templated return type should not be complained about

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#1133]
Given a template `T` that's required to extend/implement `Foo`, the following code should be fine:

```php
interface Foo {}

interface IFooGetter {
    /** @return Foo */
    public function getFoo();
}

/**
 * @template T as Foo
 */
class FooGetter implements IFooGetter {
    /** @var T */
    private $t;

    /**
     * @param T $t
     */
    public function __construct(Foo $t)
    {
        $this->t = $t;
    }

    /**
     * @return T
     */
    public function getFoo()
    {
        return $this->t;
    }
}
```
Actual output: `ERROR: LessSpecificImplementedReturnType - 28:16 - The return type 'Foo' for IFooGetter::getFoo is more specific than the implemented return type for FooGetter::getfoo 'T'`

https://getpsalm.org/r/2875433a78

cc @aidantwoods

**Comment by aidantwoods** [comment]
This looks promising! :) Did you just add the 
```
/**
 * @template T as Foo
 */
```
annotation? I hadn't seen that before

**Comment by muglug** [comment]
Fixed last night with https://github.com/vimeo/psalm/commit/ca6149ed172585063bd13adfbddd3d34a46d220b

> Did you just add the `/** @template T as Foo */` annotation?

Nope, it has been around for a while.
